heimlig/src/hsm/keystore.rs

```diff
@@ -66,100 +67,102 @@ impl From<u32> for KeyId {
     }
 }
 
-impl From<KeyType> for u32 {
-    fn from(value: KeyType) -> Self {
+impl From<Curve> for u32 {
+    fn from(value: Curve) -> Self {
         value as u32
     }
 }
 
-impl TryFrom<u32> for KeyType {
+impl TryFrom<u32> for Curve {
     type Error = ValidationError;
 
     fn try_from(value: u32) -> Result<Self, Self::Error> {
         match value {
-            SYMMETRIC_128_BITS => Ok(Self::Symmetric128Bits),
-            SYMMETRIC_192_BITS => Ok(Self::Symmetric192Bits),
-            SYMMETRIC_256_BITS => Ok(Self::Symmetric256Bits),
-            ECC_KEYPAIR_NIST_P256 => Ok(Self::EccKeypairNistP256),
-            ECC_KEYPAIR_NIST_P384 => Ok(Self::EccKeypairNistP384),
+            crate::integration::raw_jobs::NIST_P256 => Ok(NistP256),
+            crate::integration::raw_jobs::NIST_P384 => Ok(NistP384),
             _ => Err(ValidationError::InvalidValue),
         }
     }
 }
 
+impl Curve {
+    pub const fn size(&self) -> usize {
```

Comment:
I added match statements over the specific curve values. That way, we can have special handling of curves that behave differently.

Reply:
> Does it make sense to return the size in bits? E.g. for NIST P-521.
I would like to avoid that unless we really need to. Currently, all sizes are in bytes.